compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -224,6 +227,46 @@ internal class BindingModuleGenerator(
     }
   }
 
+  private fun checkNotGeneric(
+    type: ClassDescriptor,
+    boundTypeDescriptor: ClassDescriptor
+  ) {
+    if (boundTypeDescriptor.declaredTypeParameters.isNotEmpty()) {
+
+      fun StringBuilder.describeTypeParameters(type: KotlinType): StringBuilder {
+        if (type.arguments.isNotEmpty()) {
+          append("<")
+
+          for ((count, typeArgument) in type.arguments.withIndex()) {
+            if (count > 0) {
+              append(", ")
+            }
+            append(DescriptorUtils.getClassDescriptorForType(typeArgument.type).name)
+            describeTypeParameters(typeArgument.type)
+          }
+
+          append(">")
+        }
+
+        return this
+      }
+
+      val actualBoundType = type.typeConstructor
+          .supertypes
+          .first { DescriptorUtils.getClassDescriptorForType(it) == boundTypeDescriptor }
```

Comment:
Use the extension function: `it.classDescriptorForType()`

Reply:
Ah, damn. It was already there and used in the very file... Thanks!